common/lib.php

```diff
@@ -85,15 +85,15 @@ function equella_getssotoken($course = null) {
             // see if the user has a role that is linked to an equella role
             $shareid = $CFG->{"equella_{$role->shortname}_shareid"};
             if (!empty($shareid)) {
-                return equella_getssotoken_raw($USER->username, $shareid, $CFG->{"equella_{$role->shortname}_sharedsecret"});
+                return equella_getssotoken_raw($USER->equellauser, $shareid, $CFG->{"equella_{$role->shortname}_sharedsecret"});
```

Comment:
Can you do `$CFG -> equellauser` ?  I just compared this one with other settings and they are all access from $CFG.

Reply:
The config $CFG is not reloaded when you update the settings.
That's why I call the method mod_equella_after_config() 
Moreover, we need to keep the username if the default value is selected. Hence why I add an extra step to store the data as part of $USER